[All hosts](security docs refresh) (#3010)
OfficeDev/office-js-docs-pr: docs/develop/use-sso-to-get-office-signed-in-user-token.md

   - `d3590ed6-52b3-4102-aeff-aad2292ab01c` (Microsoft Office)
   - `ea5a67f6-b6f3-4338-b240-c655ddc3cc8e` (Microsoft Office)
   - `57fb890c-0dab-4253-a5e0-7188c88b2bb4` (Office on the web)
   - `08e18876-6177-487e-b8b5-cf950c1e598c` (Office on the web)
   - `bc59ab01-8403-45c6-8796-ac3ef710b3e3` (Outlook on the web)

   For each ID, take these steps:

   a. Select **Add a client application** button and then, in the panel that opens, set the `[app-id-guid]` to the Application (client) ID and check the box for `api://localhost:44355/[app-id-guid]/access_as_user`.

   b. Select **Add application**.

1. Select **API permissions** under **Manage** and select **Add a permission**. On the panel that opens, choose **Microsoft Graph** and then choose **Delegated permissions**.

1. Use the **Select permissions** search box to search for the permissions your add-in needs. Search for and select the **profile** permission. The `profile` permission is required for the Office application to get a token to your add-in web application.

   - profile

   > [!NOTE]

>    > The `User.Read` permission may already be listed by default. It is a good practice not to ask for permissions that are not needed, so we recommend that you uncheck the box for this permission if your add-in does not actually need it.

Reviewer: @lindalu-MSFT I think this change implies that they _never_ need User.Read. But sometimes they do.

1. Select the **Add permissions** button at the bottom of the panel.

1. On the same page, choose the **Grant admin consent for \<tenant-name\>** button, and then select **Yes** for the confirmation that appears.

## Create the Office Add-in

> # [Visual Studio 2019](#tab/vs2019)

Reviewer: VS 2022 is now GAed.

1. Start Visual Studio 2019 and choose to **Create a new project**.

> 1. Search for and select the **Excel Web Add-in** project template. Then choose **Next**. Note: SSO works with any Office application, but for this article will work with Excel.

Reviewer 2: ```suggestion
1. Search for and select the **Excel Web Add-in** project template. Then choose **Next**. Note: SSO works with any Office application, but for this article we use Excel.
```

   async function getIDToken() {
     try {
       let userTokenEncoded = await OfficeRuntime.auth.getAccessToken({
         allowSignInPrompt: true,
       });
       let userToken = jwt_decode(userTokenEncoded);
       document.getElementById("userInfo").innerHTML =
         "name: " +
         userToken.name +
         "<br>email: " +
         userToken.preferred_username +
         "<br>id: " +
         userToken.oid;
       console.log(userToken);
     } catch (error) {
       document.getElementById("userInfo").innerHTML =
         "An error occurred. <br>Name: " +
         error.name +
         "<br>Code: " +
         error.code +
         "<br>Message: " +
         error.message;
       console.log(error);
     }
   }
   ```

1. Save the file.

---

## Run the add-in

# [Visual Studio 2019](#tab/vs2019)

1. Choose **Debug > Start Debugging**, or press **F5**.

# [yo office](#tab/yooffice)

Run `npm start` from the command line.

---

1. When Excel starts, sign in to Office with the same tenant account you used to create the app registration.
1. On the **Home** ribbon, choose **Show Taskpane** to open the add-in.
1. In the add-in's task pane, choose **Get ID token**.

> The add-in will display the name, email, and ID of the account you signed in with.

Reviewer 2: ```suggestion
The add-in displays the name, email, and ID of the account you signed in with.
```
This result should be a paragraph under the last step so I added 4 spaces to indent. You can remove the blank line before it, for clarity.
